spyder/api/preferences.py

```diff
@@ -56,3 +59,32 @@ def setup_page(self):
         self.setLayout(layout)
         """
         raise NotImplementedError
+
+    def prompt_restart_required(self):
```

Comment:
This is not the right approach @steff456. We're repeating here the same method present in `GeneralConfigPage`. Since that class and `PluginConfigPage` both inherit from `SpyderConfigPage`, let's move this method to that class instead, so the child classes have access to it.